module/applications/dice/roll-configuration-dialog.mjs

```diff
@@ -0,0 +1,295 @@
+     * @param {number} index                   Index of the roll within all rolls being prepared.
+     */
+    Hooks.callAll("dnd5e.buildRollConfig", this, config, formData, index);
+
+    if ( formData?.get("situational") && (config.situational !== false) ) {
+      config.parts.push("@situational");
+      config.data.situational = formData.get("situational");
+    }
+
+    return config;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Make any final modifications to rolls based on the button clicked.
+   * @param {string} action  Action on the button clicked.
+   * @returns {BasicRoll[]}
+   * @protected
+   */
+  _finalizeRolls(action) {
+    return this.rolls;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Rebuild rolls based on an updated config and re-render the dialog.
+   */
+  rebuild() {
+    this._onChangeForm(this.options.form, new Event("change"));
+  }
+
+  /* -------------------------------------------- */
+  /*  Event Handling                              */
+  /* -------------------------------------------- */
+
+  /**
+   * Handle submission of the dialog using the form buttons.
+   * @param {Event|SubmitEvent} event    The form submission event.
+   * @param {HTMLFormElement} form       The submitted form.
+   * @param {FormDataExtended} formData  Data from the dialog.
+   * @private
+   */
+  static async #handleFormSubmission(event, form, formData) {
+    const rolls = this._finalizeRolls(event.submitter?.dataset?.action);
+    await this.close({ dnd5e: { rolls } });
+  }
+
+  /* <><><><> <><><><> <><><><> <><><><> */
+
+  /** @inheritDoc */
+  _onChangeForm(formConfig, event) {
+    super._onChangeForm(formConfig, event);
+
+    const formData = new FormDataExtended(this.element);
+    if (formData.has("rollMode")) this.message.rollMode = formData.get("rollMode");
+    this.rolls = this.#buildRolls(foundry.utils.deepClone(this.config), formData);
+    this.render({ parts: ["formulas"] });
+  }
+
+  /* <><><><> <><><><> <><><><> <><><><> */
+
+  /** @override */
+  _onClose(options = {}) {
+    this.options.resolve?.(options[game.system.id]?.rolls ?? []);
+  }
+
+  /* -------------------------------------------- */
+  /*  Constructor                                 */
+  /* -------------------------------------------- */
+
+  /**
+   * A helper to handle displaying and responding to the dialog.
+   * @param {BasicRollProcessConfiguration} [config]        Initial roll configuration.
+   * @param {BasicRollConfigurationDialogOptions} [dialog]  Dialog configuration options.
+   * @param {BasicRollMessageConfiguration} [message]       Message configuration.
+   * @returns {Promise}
+   */
+  static async configure(config={}, dialog={}, message={}) {
+    return new Promise((resolve, reject) => {
+      new this(config, message, { ...(dialog.options ?? {}), resolve, reject }).render(true);
+    });
```

Comment:
If you take my suggestion above to change the contract for `_finalizeRolls` and not propagate the final rolls via the `close` options, then we could write this in a slightly more idiomatic AppV2 style, removing the need for `resolve` and `reject` options in `BasicRollConfigurationDialogOptions`, and the `_onClose` override.

```suggestion
    return new Promise(resolve => {
      const app = new this(config, message, dialog);
      app.addEventListener("close", () => resolve(app.rolls), { once: true });
      app.render({ force: true });
    });
```